Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -384,19 +384,20 @@ struct MacroSystem {
 // MARK: - MacroApplication
 
 /// Removes attributes from a syntax tree while maintaining their surrounding trivia.
-private class AttributeRemover: SyntaxRewriter {
-  var attributesToRemove: [AttributeSyntax]
+@_spi(Testing)
+public class AttributeRemover: SyntaxRewriter {
+  let predicate: (AttributeSyntax) -> Bool
 
   var triviaToAttachToNextToken: Trivia = Trivia()
 
-  init(attributesToRemove: [AttributeSyntax]) {
-    self.attributesToRemove = attributesToRemove
+  public init(where predicate: @escaping (AttributeSyntax) -> Bool) {
+    self.predicate = predicate
```

Comment:
Could you add a doc comment here that say that all the attributes for which the predicate returns `true` will be removed just to clarify that it’s not other way round.